src/core/network/qgsnetworkaccessmanager.h

```diff
@@ -227,6 +227,20 @@ class CORE_EXPORT QgsNetworkAuthenticationHandler
      */
     virtual void handleAuthRequest( QNetworkReply *reply, QAuthenticator *auth );
 
+    /**
+     * Called to initiate a network authentication through external browser \a url.
+     *
```

Comment:
I think you should add a warning in these two methods that they aren't guaranteed to be called from the main thread, and may be called by a background thread.